plugins/node/instrumentation-dataloader/src/instrumentation.ts

```diff
@@ -0,0 +1,248 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import {
+  diag,
+  trace,
+  context,
+  Link,
+  SpanStatusCode,
+  SpanKind,
+} from '@opentelemetry/api';
+import { DataloaderInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import type * as Dataloader from 'dataloader';
+
+const MODULE_NAME = 'dataloader';
+
+type DataloaderInternal = typeof Dataloader.prototype & {
+  _batchLoadFn: Dataloader.BatchLoadFn<unknown, unknown>;
+  _batch: { spanLinks?: Link[] } | null;
+};
+
+type LoadFn = typeof Dataloader.prototype['load'];
+type LoadManyFn = typeof Dataloader.prototype['loadMany'];
+
+export class DataloaderInstrumentation extends InstrumentationBase {
+  constructor(config: DataloaderInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-dataloader', VERSION, config);
+  }
+
+  protected init() {
+    return [
+      new InstrumentationNodeModuleDefinition<typeof Dataloader>(
+        MODULE_NAME,
+        ['^2.0.0'],
+        (dataloader, moduleVersion) => {
+          diag.debug(`Applying patch for ${MODULE_NAME}@${moduleVersion}`);
+
+          this._patchLoad(dataloader.prototype);
+          this._patchLoadMany(dataloader.prototype);
+
+          return this._getPatchedConstructor(dataloader);
+        },
+        (dataloader, moduleVersion) => {
+          diag.debug(`Removing patch for ${MODULE_NAME}@${moduleVersion}`);
+
+          if (isWrapped(dataloader.prototype.load)) {
+            this._unwrap(dataloader.prototype, 'load');
+          }
+
+          if (isWrapped(dataloader.prototype.loadMany)) {
+            this._unwrap(dataloader.prototype, 'loadMany');
+          }
+        }
+      ),
+    ];
+  }
+
+  override getConfig(): DataloaderInstrumentationConfig {
+    return this._config;
+  }
+
+  override setConfig(config: DataloaderInstrumentationConfig) {
+    this._config = config;
+  }
+
+  private shouldCreateSpans(): boolean {
+    const config = this.getConfig();
+    const hasParentSpan = trace.getSpan(context.active()) !== undefined;
+    return hasParentSpan || !config.requireParentSpan;
+  }
+
+  private _getPatchedConstructor(
+    constructor: typeof Dataloader
+  ): typeof Dataloader {
+    const prototype = constructor.prototype;
+    const self = this;
+
+    function PatchedDataloader(
+      ...args: ConstructorParameters<typeof constructor>
+    ) {
+      const inst = new constructor(...args) as DataloaderInternal;
```

Comment:
`_batchLoadFn` isn't on the prototype?

Reply:
It is an instance variable that gets set as part of the constructor with the argument passed in: https://github.com/graphql/dataloader/blob/main/src/index.js#L64